### PR TITLE
Update to openapi-generator compiled code

### PR DESCRIPTION
**Pull request by silasbw**
```
USERNAME=kubernetes ../gen/openapi/typescript.sh src settings
```

### CODE REVIEW

**Comment by silasbw** [comment]
Results from https://github.com/kubernetes-client/gen/pull/125.

**Comment by silasbw** [comment]
Related issue: https://github.com/kubernetes-client/javascript/issues/290

**Comment by drubin** [comment]
Thanks for this! 

I tried looking through this but its almost impossible to review because of the number of file changes. Is it possible to split the code changes into multiple commits splitting the autogenerated files and the manual changed configuration/setup files?

Also wondering if it's possible to put the auto-generated code into its own folder?

**Comment by silasbw** [comment]
I split out the manual config changes here: https://github.com/kubernetes-client/javascript/pull/307

The auto-generated code has this layout with OUTPUT=src:

```
OUTPUT/api.ts
OUTPUT/api/*.ts
OUTPUT/model/*.ts
```

Were you thinking something like OUTPUT=src/gen?

**Comment by drubin** [comment]
>Were you thinking something like OUTPUT=src/gen? 

I think so it makes it much easier to know the entire directory is auto-generated.

Ruby puts it under `lib/` [see](https://github.com/kubernetes-client/ruby/tree/master/kubernetes/lib)  but I don't think this was intentional.

Maybe @brendanburns  can weigh in here, I kinda think lib might make more sense than `src/gen`

File: .prettierignore [hunk 1]
```diff
@@ -1 +1,3 @@
 src/api.ts
```
Comment (brendandburns) [inline .prettierignore]: delete this since we're deleting the file.

File: package.json [hunk 2]
```diff
@@ -28,6 +28,8 @@
     ],
     "exclude": [
       "src/api.ts",
```
Comment (brendandburns) [inline package.json]: delete this line since we're deleting the file.

**Comment by brendandburns** [comment]
I don't feel super strongly. we've used `gen` in other places. @drubin if you feel strongly let's switch to `lib` otherwise, we can leave it alone.

**Comment by brendandburns** [comment]
Rather than splitting this into multiple PRs, I think I'd prefer it to just be multiple commits in the same PR.

That way you can read each commit differently (or not read it in the case of the generated code) but we can still merge it all at once.

Thanks!

(and thanks for doing this!)

**Comment by silasbw** [comment]
Re single PR, multiple commits: sounds good, I'll do that.

I'll generate code in the output directory `gen/`. It should be easy for me or someone else to change the output directory.

**Comment by silasbw** [comment]
https://github.com/kubernetes-client/javascript/pull/307 implements the changes discussed so far in this PR.

**Comment by drubin** [comment]
No strong feelings @brendanburns and @silasbw

**Pull request closed without merging**